brian2/core/variables.py

```diff
@@ -957,15 +966,9 @@ def set_item(self, item, value, level=0, namespace=None):
             )
         elif isinstance(item, str):
             try:
-                float(value)  # only checks for the exception
-                try:
-                    # length-1 arrays are also convertible to float, but we
-                    # don't want the repr used later to be something like
-                    # array([...]).
-                    value = value[0]
-                except (IndexError, TypeError):
-                    # was scalar already apparently
-                    pass
+                if isinstance(value, str):
+                    raise TypeError  # Will be dealt with below
+                value = np.asanyarray(value).item()
```

Comment:
Not sure I understand these changes, can you summarise?

Reply:
This change is a slightly more complex issue related to what I wrote in my earlier comment: in future versions, we can no longer use `float(x)` as a check that `x` is either a scalar number or a length-1 array with a number (and not for example a string) – it will wail on length-1 arrays.